Yoast/javascript
# Fixes parsing of paragraphs within headings

## Summary

This PR can be summarized in the following changelog entry:

* _[not-user-facing]_ Fixes the parsing of paragraphs within headings when using the tree parser. Previously, it crashed the building of the tree.

## Relevant technical choices:

* This PR refactors the building of the tree. It simplifies it by quite a bit.
    * Main change is that `Node`s within paragraphs or headings are not transformed to `FormattingElement`s anymore, but added as formatting as-is. We already do this with `Ignored` nodes (see #70 ).

## Test instructions

This PR can be tested by following these steps:

* Start and open the example.
* Insert a HTML paragraph within a heading. 
   E.g.:
   ```html
   <h2>This is <p>some paragraph</p> within a heading</h2>
   ```
* Make sure that the _"Build the tree"_ toggle is turned on.
    * Can be found in the _Controls_ section of the example, near the end of the page.
* Check the tree that has been built (in the _Tree_ section, under _Controls_).
   * It should have a paragraph added under the `formatting` of the `Heading`'s `TextContainer`.
   * The heading's content should contain the entire text (e.g. "This is some paragraph within a heading." in the example above).
   * The paragraph should not have any content itself. 

Fixes #61
